src/vcpkg/export.ifw.cpp

```diff
@@ -15,6 +15,86 @@ namespace vcpkg::Export::IFW
     using Dependencies::ExportPlanType;
     using Install::InstallDir;
 
+    std::string safe_rich_from_plain_text(StringView text)
+    {
+        // looking for `&`, not followed by:
+        // - '#<numbers>;`
+        // - '#x<hex numbers>;`
+        // - `<numbers, letters, or _>;`
+        // (basically, an HTML character entity reference)
+        using P = Parse::ParserBase;
+        constexpr static auto is_word = [](char ch) { return P::is_alphanum(ch) || ch == '_'; };
+        constexpr static auto is_hex_digit = [](char ch) {
+            return P::is_ascii_digit(ch) || (ch >= 'A' && ch <= 'F') || (ch >= 'a' && ch <= 'f');
+        };
```

Comment:
I would like to see these put next to is_alphanum et al. (And, for that matter, taken out of being members of ParserBase because I think these functions are fairly universial)

The latter change is probably unrelated to this PR but adding new ones should probably be done in the same place. (I could have swore we also already had is_hexit or something like that somewhere?)